reference genomes 


sequencing ---> mapping to reference genomes

de novo assembly take the contigs and overlap on the base of similar regions. 

metagenomic binning to generate metagenome-assembled genomes. 


Workflow:
\begin{itemize}
	\item mapping to refernce genomes: high troughput and need reference
	\item de novo assembly automatic assembly is high throughput, while instead manually curated assembly and binning is done with Anvi'o, low throughput but accurate
\end{itemize}


Anvi'o is , developed by the community


\subsection{De novo assembly and binning}
if with a series of genomes, shor reads, assemble in contis, and assemble them in MAGs through ...


the binning process, with a series of contigs. 

visually cluster

differential coverage is how good a metagenome in a contig
sequence composition k mer sequences, sequence from the same organism same

Imagine having 2 contigs. coverage associated to contigs. differential coverage are represented in different abundances in samples. 
when looking to the coverage, related to the abundance of the sample


look to differential coverage in table. spot covariation, it is possible ot make some sort of clustering.

\begin{figure}[h]
\caption{}
\centering
\includegraphics[width=0.6\textwidth]{}
\label{}
\end{figure}


K-mer frequenceis is usef for sequence composition. 
In the case of dimers: look to repetitions of two nucleotides. generate a table with occurrences with the 2 nucleotide sequneces. the k-mer count has to be done in both the directions. 

put the information together, sum up the counts from the two cases. 

do the k-mer table for each contig. sequence x and l tend to covariate. we can so with some reasoning group the ocntigs and asay that they come from the same genome.

import the taxonomic annotations


graph of anvi'o
representation 
each radius  is a contig, contig assembled in bin.
differential. together in a bin
select a bin, export it and run the commands on the slide.  