\graphicspath{{chapters/images/04/}}
\chapter{Sequencing data}

\section{Choosing the optimal technology}
When performing a genetics or genomics study it is best to be as hypothesis driven as possible and use already available data to guide the new analysis.
Moreover to choose the optimal sequencer the parameters that need to be considered are:

\begin{multicols}{2}
	\begin{itemize}
		\item Throughput.
		\item Cost.
		\item Read lengths.
		\item Data output (reads per run).
		\item Coverage.
		\item Sequencing errors (indel, substitution, CG deletion, AT bias).
		\item Library preparation compatibility.
		\item Speed (run time).
	\end{itemize}
\end{multicols}

	\subsection{Comparing different sequencing technologies}

	\begin{multicols}{2}
		\begin{description}
			\item[Illumina NovaSeq]: optimal for sequencing a lot of DNA molecules at the same time like in the case of genomes or metagenomes.
				It can’t go over $300 bp$ readlines run, but it has the highest throughput so far.
				It is capable of multiplexing, differentiating the different samples with an unique barcode.
			\item[Illumina iSeq]: optimal for sequencing shorter genomes.
			\item[NanoPore (minion)]: it is a pocket-sized wet-lab free sequencer for DNA, RNA and (possibly) proteins, but the read lengths is smaller than Illumina's.
				The machine is cheap;, but the running flow is more expensive over time.
				It's a real-time sequencer.
			\item[PacBio]: has very long reads but carries a high amount of error.
		\end{description}
	\end{multicols}

	A solution to reduce the impact of error or weaknesses of one of the sequencers is to use more than one for the same project.
	Consider there is a need to sequence the genome of a bacterium: PachBio would give a lot of sequence errors, while Illumina wuild be unable to reconstruct the sample due to assembly ambiguities.
	In the end PacBio will construct the genome and Illumina will correct sequence errors.
	This solution doesn't work in complex sample with more than one genome, because there is no way to a priori which reads are coming from one organism.
	Another widely adopted solution is to sequence multiple times one molecule so to reduce random errors.
	This is effective but does not resolve systematic error like the one of PacBio with homopolymers.

	\subsection{Sequencers' output}
	All sequencing platforms translate the physical read signal into files in FASTQ format.
	This files contain the sequencing reads and the quality of each base.

\section{Base callers}
A base caller is an algorithm that translates the analogical signal of the reading into numbers and nucleotides.
The most popular algorithm is Phred.
Phred tries to correct errors derived from the sequencing reaction and electrophoresis.
It was tested on a huge dataset of gold standard sequences (finished human and C. elegans sequences generated by highly-redundant sequencing).
Its results were compared with the traditional ABI base caller and Phred was considerably more accurate with $40$-$50\%$ fewer errors.
This algorithm need to be able to understand when it is impossible to recover an high quality sequencing and so it needs to be able to give up for low-quality reads.
The confidence that the base caller has to call a certain nucleotide ATCG is annotated in the FASTQ file, allowing for quality control downstream.

	\subsection{Errors solved by Ilumina's base caller}

	\begin{multicols}{2}
		\begin{description}
			\item[Phasing noise $\phi$]: when a certain base is not seen frequently, the first time in which it will reappear there will be a spike in the graph, increasing the signal of the nearby bases.
				This will cause errors in estimating the real nucleotide that is occurring in the site.
				This problem can be solved by waiting more time between readings, but the sequencing will be less efficient and the throughput will be lower.
				The machine needs to find a trade-off between efficiency and clear reading.
			\item[Signal decay $\delta$]: after a while the sequencer has read the same base, or the same repeated couple of bases, the signal will go down and at some point will be indistinguishable.
				At some point you will need to cut the read since it will be not trustable after a while.
			\item[Mixed cluster $\mu$]: two different fragments can enter the same cluster and the sequencer will read the two signals simultaneously.
			\item[Boundary effects $\omega$]:  in this case the machine needs to interpret an image and it cannot distinguish between the signal and the background.
			\item[Cross-talk $\mathcal{X}$]
			\item[Fluophore accumulation $\tau$]
		\end{description}
	\end{multicols}

	\subsection{Density on the flow cell}
	The density of the flow cell is the number of cluster in it.
	Under clustering will reduce the sequencer throughput, while over clustering will cause errors due to the limited resolution of the reading of the bases.
	There is a need to find the optimal number of cluster that maximises the throughput without introducing overlapping in the reading of clusters.

	\subsection{An ecology of base callers}
	Base callers need to find a satisfying trade-off between accuracy and computational efficiency.
	The quality is the estimation of the probability that the nucleotide in a certain position is correct.
	The PHRED score reported by the base caller and the PHRED score of mapping to a reference sequence are different.
	Base callers need to be calibrated with a standard to make sure that the estimation of the quality is accurate enough.

\section{FASTQ format}

	\subsection{Composition}
	The FASTQ format is composed by:

	\begin{multicols}{2}
		\begin{enumerate}
			\item ‘@’ followed by a sequence identifier.
				This identifier contains the unique instrument name, the flowcell and tile number, the x and y coordinates of the cluster within the tile, the index number for multiplexing and the pair number of paired-end sequencing.
				In Illumina MiSeq, each flowcell has 8 microfluidic channels (lanes), each lane contains three columns with 96 tiles, and can sequence up to 96 multiplexed samples.
			\item The sequence. It could be mate paired for paired end sequencing.
			\item ‘+’, optionally followed by a sequence Identifier.
			\item The quality scores.
				Quality is a number based on the estimated probability of error.
				$p$=probability  of error, $Q = -10 \dot p$ .
				A base quality of at least $20$ is needed to reach  $1\%$ of error.
				A base quality of $40$ means the probability of error is $0.01\%$.
				The FASTQ quality score is the phred score +33, converted in CHAR code.
		\end{enumerate}
	\end{multicols}

	The FASTA file format is a FASTQ fomrat without the quality score reported and the seq ID is preceded by '>'.

	\subsection{Quality control: read length distribution}
	Quality scores are typically used to perform quality control and cleaning of the reads.
	This result in a FASTA output file to be used downstream.
	However there are algorithms that can use directly FASTQ files, performing autonomously the cleaning and quality control of the reads.
	The quality score is an indication of how well the sequencing run went.
	Typically the quality decreases when the read length increases due to the fact that sequencers have problem when are run in continuum.
	A solution to this problem is to cut the reads when the quality becomes too low.
	Another problem happens when the adapter is included into the read.
	This happens typically with short reads and a solution is to cut it and a part of the sequence.
	FastQC can be used to plot the quality distribution of the data.
	Another way to asses read quality is to consider the average quality of the entire read, discarding the low-quality ones.


	\subsection{Duplication artifacts}
	It is not frequent to see duplication, but it can be a problem especially when there is a need to quantify gene expression or copy number of genes in a bacterial genome.
	The distribution of the duplicates should be the same of the distribution of the reads.
	There should’t be any bias along the length of the reads, but if there is it should be due to a repetition of sequencing of the same read or when the adapter and primer have been read.

	\subsection{GC content analysis}
	Each organism has a signature GC content, so when plotting it a normal distribution is expected.
	Multiple peaks are an indicator of the presence of two different organisms.

	\subsection{K-mers frequency plot}
	K-mer frequencies are a way to catch systematic sequencing error: when mapping a genome it is now the relative frequency of each K-mer.
	That can be compared with the K-mers frequencies to catch systematic errors in sequencing and to assess its quality.
	Frequent k-mers can be a signature, as the GC content.
	The expected coverage of a k-mer with reads of length L:

	$$L_{cov} = \frac{L-k+1}{L} \, \times \, Cov$$

	Then, given a k-mer, it can be seen in how many reads it is present.
	For a typical K-mer its coverage should be around $40\%$.
	Coverage higher than $80\%$ should indicate that this k-mer is located in more than one position.
	Other values arises from errors.

	\subsection{Low-complexity artefacts}
	Same nucleotide repeats (especially A) are in a lot of cases artefacts.
	This is due to systematic error and can be evidenced through quality control.
	To measure if these sequence are in fact artefacts parameters to take into consideration are:

	\begin{multicols}{2}
		\begin{itemize}
			\item Low complexity.
			\item Low entropy.
			\item High compression (the artefacts increase the information inside the file).
		\end{itemize}
	\end{multicols}

	But some low-complexity sequences are not artefacts:

	\begin{multicols}{2}
		\begin{itemize}
			\item Hydrophobic transmembrane alpha-helical sequences in membrane proteins.
			\item CAG repeats in genes causing Huntington disease, spinal and bulbar muscular atrophy, dentatorubropallidoluysian atrophy.
			\item Proline-rich regions in proteins.
			\item Poly-A tails in nucleotide sequence.
			\item Micro-satellites.
		\end{itemize}
	\end{multicols}

	\subsection{FASTQ quality control (QC)}
	FASTQ quality control is the first step in any NGS pipeline.
	It consists of:

	\begin{multicols}{2}
		\begin{description}
			\item[Clipping/trimming]: removing (low quality) parts of reads.
			\item[Masking]: avoiding to consider parts of reads that can have low entropy for example.
			\item[Read removal]: discard low quality reads or reads that are too short after clipping.
		\end{description}
	\end{multicols}

	Additional features that can be exploited for QC are the GC content, clustering for contamination detection, TAG identification,ambiguous bases.
